File: wishlist-service/src/main/java/com/smartcart/wishlistservice/service/WishlistService.java
package com.smartcart.wishlistservice.service;

import com.smartcart.wishlistservice.entity.Wishlist;
import com.smartcart.wishlistservice.repository.WishlistRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class WishlistService {

    private final WishlistRepository wishlistRepository;

    public Wishlist addToWishlist(Long userId, Long productId) {
        if (wishlistRepository.findByUserIdAndProductId(userId, productId).isPresent()) {
            throw new RuntimeException("Product already in wishlist");
        }
        Wishlist wishlist = Wishlist.builder()
                .userId(userId)
                .productId(productId)
                .build();
        return wishlistRepository.save(wishlist);
    }

    public void removeFromWishlist(Long userId, Long productId) {
        Wishlist wishlist = wishlistRepository.findByUserIdAndProductId(userId, productId)
                .orElseThrow(() -> new RuntimeException("Item not found in wishlist"));
        wishlistRepository.delete(wishlist);
    }

    public List<Wishlist> getUserWishlist(Long userId) {
        return wishlistRepository.findByUserId(userId);
    }
}

File: frontend/src/services/wishlistService.js
import api from './axiosConfig';

export const addToWishlist = (userId, productId) => {
    return api.post(`/wishlist/${userId}/add?productId=${productId}`);
};

export const removeFromWishlist = (userId, productId) => {
    return api.delete(`/wishlist/${userId}/remove/${productId}`);
};

export const getWishlist = (userId) => {
    return api.get(`/wishlist/${userId}`);
};
